### PR TITLE
Prisma set initial set up

### PR DESCRIPTION
**Pull request by ryanhtang**
Did the initial setup for Prisma. The prisma schema (schema.prisma) is not fully set up because the setup is based on what is in our database/schema. Right now it contains a filler model "user" because it needed at least one model to be set up. I

### CODE REVIEW

**Review by ray-sjsu — COMMENTED** [review]
We could use postgres, but we do need a server. Are we hosting postgres locally or a cloud database?

**Comment by Luceium** [comment]
@alia49 was working on using mongodb. We can host it on mobgodb atlas.

**Comment by ray-sjsu** [comment]
@Luceium we could still use [Prisma](https://www.prisma.io/docs/orm/overview/databases/mongodb) as an ORM to connect MongoDB with Nextjs. It would be easy to update models and the database schema from Prisma.
@ryanhtang what do you think?

**Comment by Luceium** [comment]
That was the plan already

**Comment by ryanhtang** [comment]
> @Luceium we could still use [Prisma](https://www.prisma.io/docs/orm/overview/databases/mongodb) as an ORM to connect MongoDB with Nextjs. It would be easy to update models and the database schema from Prisma. @ryanhtang what do you think?

Sorry for the late response but yeah I think this is what we are going to do using prisma.

File: prisma/schema.prisma [hunk 1]
```diff
@@ -1,39 +1,41 @@
...
 
 model User {
-  id    String  @id @default(dbgenerated()) @map("_id")
+  id    String  @id @default(auto()) @map("_id") @db.ObjectId
   email String  @unique
   name  String?
   posts Post[]
 }
 
 model Post {
```
Comment (Luceium) [inline prisma/schema.prisma:17]: What are posts?
Are we supporting this in any of our releases or is this optional?

File: prisma/schema.prisma [hunk 2]
```diff
@@ -1,39 +1,41 @@
...
-// Example on how to use Prisma in API routes
-// import { PrismaClient } from '@prisma/client';
-
-// const prisma = new PrismaClient();
+model Hotel {
+  id            String   @id @default(auto()) @map("_id") @db.ObjectId // Use ObjectId for MongoDB
+  name          String
+  location      Json
+  amenities     String[]
+  pricePerNight Int
```
Comment (Luceium) [inline prisma/schema.prisma:30]: Depends on the room selected so maybe this should be a range

File: prisma/schema.prisma [hunk 3]
```diff
@@ -1,39 +1,41 @@
...
+  createdAt     DateTime @default(now())
+}
 
-// export default async function handler(req, res) {
-//   const users = await prisma.user.findMany();
-//   res.status(200).json(users);
-// }
+model Ratings {
+  id             String @id @default(auto()) @map("_id") @db.ObjectId // Use ObjectId for MongoDB
+  roomsAvailable Int
```
Comment (Luceium) [inline prisma/schema.prisma:37]: This should be under Hotel

File: prisma/schema.prisma [hunk 4]
```diff
@@ -1,39 +1,41 @@
...
+  description    String
```
Comment (Luceium) [inline prisma/schema.prisma:38]: should this be in ratings

**Review by Luceium — COMMENTED** [review]
I left a few comments.
Overall, this PR is shaping up.
Once this work is done please move on to creating server actions to get the values desired.

**Pull request closed without merging**